pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -309,7 +317,14 @@ func (c *Reconciler) resolvePipelineState(
 		// We need the TaskRun name to ensure that we don't perform an additional remote resolution request for a PipelineTask
 		// in the TaskRun reconciler.
 		trName := resources.GetTaskRunName(pr.Status.TaskRuns, pr.Status.ChildReferences, task.Name, pr.Name)
-		fn := tresources.GetTaskFunc(ctx, c.KubeClientSet, c.PipelineClientSet, c.resolutionRequester, pr, task.TaskRef, trName, pr.Namespace, pr.Spec.ServiceAccountName)
+
+		// get verificationpolicies for trusted resources
+		vp, err := getVerificationPolicies(ctx, c.verificationPolicyLister, pr.Namespace)
```

Comment:
it seems like this function is being called twice in each reconcile, why is that? is there any way to avoid it?

Reply:
It's called twice because we need to fetch the policy for pipeline and the task in pipelinerun reconciler. It is possible to pass but it will require function signature change to `reconcile.` I wonder if it is necessary to do so.

Reply:
Since this is a alpha feature, I don't lean to change the function and pass it all the way down to the second place

Reply:
ah okay makes sense - and if we use the lister then there is no actual API call required so that should be fine 👍 

it might make your function signatures a bit simpler to pass around the objects vs. the lister functions but i dont feel too strongly